### PR TITLE
[Task 22] User List Interactor Unit Tests

### LINKED ISSUE #118: [Task 22] User List Interactor Test

### PR DESCRIPTION
**Pull request by candidcalcifer**
What: added unit tests for user list interactor 

Why: Need test coverage to make sure code and classes are working properly

Implementation Details: two unit tests for typical use cases (more than one ID in list, inputted userId is in list).

### CODE REVIEW

**Review by ABunchOfAtom — DISMISSED** [review]

File: app/src/test/java/com/courseproject/tindar/usecases/userlist/UserListInteractorUnitTest.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+        @Override
+        public ArrayList<String> getAllOtherUserIds(String userId) {
+            return USERLIST;
+        }
+    }
+
+    @Test
+    public void getUserListOne() {
+        MockUserListDsGateway userListDsGateway = new MockUserListDsGateway();
```
Comment (jenniferjinyoungyang) [inline app/src/test/java/com/courseproject/tindar/usecases/userlist/UserListInteractorUnitTest.java:26]: DsGateway should be used as an input argument to instantiate the interactor. and the method in the interactor should be called. Due to lack of time, I will just make a direct contribution to this PR.

File: app/src/test/java/com/courseproject/tindar/usecases/userlist/UserListInteractorUnitTest.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        assertEquals(USERLIST, userListDsGateway.getAllOtherUserIds(USER_ID_1));
+    }
+
+    @Test
+    public void getUserListTwo() {
+        MockUserListDsGateway userListDsGateway = new MockUserListDsGateway();
+        assertEquals(USERLIST, userListDsGateway.getAllOtherUserIds(USER_ID_2));
```
Comment (jenniferjinyoungyang) [inline app/src/test/java/com/courseproject/tindar/usecases/userlist/UserListInteractorUnitTest.java:33]: DsGateway should be used as an input argument to instantiate the interactor. and the method in the interactor should be called. Due to lack of time, I will just make a direct contribution to this PR.

File: app/src/test/java/com/courseproject/tindar/usecases/userlist/UserListInteractorUnitTest.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        @Override
+        public ArrayList<String> getAllOtherUserIds(String userId) {
+            return USERLIST;
+        }
+    }
+
+    @Test
+    public void getUserListOne() {
+        MockUserListDsGateway userListDsGateway = new MockUserListDsGateway();
+        assertEquals(USERLIST, userListDsGateway.getAllOtherUserIds(USER_ID_1));
```
Comment (jenniferjinyoungyang) [inline app/src/test/java/com/courseproject/tindar/usecases/userlist/UserListInteractorUnitTest.java:27]: DsGateway should be used as an input argument to instantiate the interactor. and the method in the interactor should be called. Due to lack of time, I will just make a direct contribution to this PR.

**Review by jenniferjinyoungyang — APPROVED** [review]

**Review by ABunchOfAtom — APPROVED** [review]

**Comment by navarhontes** [comment]
These tests seem rather like assuming what you're trying to prove and don't account for edge cases, but I'll approve since we don't have much time left.

**Review by navarhontes — APPROVED** [review]
These tests seem rather like assuming what you're trying to prove and don't account for edge cases, but I'll approve since we don't have much time left.
